library/std/src/sys/fs/windows/remove_dir_all.rs

```diff
@@ -90,9 +90,9 @@ fn open_link_no_reparse(
     static ATTRIBUTES: Atomic<u32> = AtomicU32::new(c::OBJ_DONT_REPARSE);
 
     let result = unsafe {
-        let mut path_str = c::UNICODE_STRING::from_ref(path);
+        let path_str = UnicodeStrRef::from_slice(path);
         let mut object = c::OBJECT_ATTRIBUTES {
-            ObjectName: &mut path_str,
+            ObjectName: path_str.as_ptr(),
```

Comment:
If I am understanding things correctly, this change only ensures nul-termination if `path` is empty but otherwise passes `path` unchanged? What is the benefit here?

Reply:
Ah I see what you mean. The slice itself doesn't contain the nul (if any). There may or may not be a nul one beyond the end of the slice but we don't know. The point is we just assume there is if some third party is hooking OS functions and requires strings be null-terminated but otherwise we assume nothing outside of the slice.

We can't really do better than that without being more invasive. The APIs themselves say nothing about null termination.